README.md

```diff
@@ -95,76 +206,71 @@ def rots(d, s, tree):
+def warp(d, s, a, b):
+  switch d:
+    case 0:
+      return swap(s ^ (a > b), a, b)
+    case _:
+      (a.a, a.b) = a
+      (b.a, b.b) = b
+      (A.a, A.b) = warp(d-1, s, a.a, b.a)
+      (B.a, B.b) = warp(d-1, s, a.b, b.b)
+      return ((A.a,B.a),(A.b,B.b))
+
+# Propagates downwards
+def down(d,s,t):
+  switch d:
+    case 0:
+      return t
+    case _:
+      (t.a, t.b) = t
+      return (rots(d-1, s, t.a), rots(d-1, s, t.b))
+
+# Swaps a single pair
+def swap(s, a, b):
+  switch s:
+    case 0:
+      return (a,b)
+    case _:
+      return (b,a)
+
+# Testing
+# -------
+
+# Generates a big tree
+def gen(d, x):
+  switch d:
+    case 0:
+      return x
+    case _:
+      return (gen(d-1, x * 2 + 1), gen(d-1, x * 2))
+
+# Sums a big tree
+def sum(d, t):
+  switch d:
+    case 0:
+      return t
+    case _:
+      (t.a, t.b) = t
+      return sum(d-1, t.a) + sum(d-1, t.b)
+
+# Sorts a big tree
 def main:
-  return render(16, demo_shader)
+  return sum(20, sort(20, 0, gen(20, 0)))
 ```
 
-And it would actually work. Even involved algorithms parallelize well on Bend.
-Long-distance communication is performed by *global beta-reduction* (as per the
-[Interaction Calculus](https://github.com/VictorTaelin/Interaction-Calculus)),
-and synchronized correctly and efficiently by
-[HVM2](https://github.com/HigherOrderCO/HVM)'s *atomic linker*.
-
+#### Benchmark
 
-- To understand the tech behind Bend, check HVM2's [paper](https://paper.higherorderco.com).
+- `bend run`: CPU, Apple M3 Max: 12.15 seconds
+- `bend run-c`: CPU, Apple M3 Max: 0.96 seconds
+- `bend run-cu`: GPU, NVIDIA RTX 4090: 0.21 seconds
 
-- Bend is developed by [HigherOrderCO](https://HigherOrderCO.com) - join our [Discord](https://discord.HigherOrderCO.com)!
+</details>
+  
+if you are interested in some other algorithms, you can check our [examples folder](https://github.com/HigherOrderCO/Bend/tree/main/examples)
 
-## Note
 
-It is very important to reinforce that, while Bend does what it was built to
-(i.e., scale in performance with cores, up to 10000+ concurrent threads), its
-single-core performance is still extremely sub-par. This is the first version of
-the system, and we haven't put much effort into a proper compiler yet. You can
-expect the raw performance to substantially improve on every release, as we work
-towards a proper codegen (including a constellation of missing optimizations).
-Meanwhile, you can use the interpreters today, to have a glimpse of what
-massively parallel programming looks like, from the lens of a Pythonish,
-high-level language!
+### Additional Resources
+ - To understand the technology behind Bend, check out the HVM2 [paper](https://docs.google.com/viewer?url=https://raw.githubusercontent.com/HigherOrderCO/HVM/main/paper/PAPER.pdf). Bend is developed by [HigherOrderCO](https://higherorderco.com/) - join our [Discord](https://discord.gg/kindelia)!
+ - Watch the [live demo video](https://x.com/i/status/1791213162525524076).
```

Comment:
There is no point in putting the live demo right at the end. Either it should be at the start or not here at all

Reply:
I am not opposed to not having it anywhere. just quoted here, but, I rather NOT